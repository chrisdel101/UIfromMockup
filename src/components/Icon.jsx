import React from "react"

function Icon(props) {
  // pass event to parent
  function handleClick(e) {
    props.onClick(e)
  }
  return (
    <div className={`icon-container-${props.i} icon-container`} onClick={handleClick}>
      <i
        className={`fa fa-${props.type} icon`}
        aria-hidden="true"
      />
    </div>
  )
}
export default Icon
