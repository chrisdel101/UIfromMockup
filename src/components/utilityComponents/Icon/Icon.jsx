import React from 'react'

function Icon(props) {
  return (
    <div className={`icon-container-${props.i} icon-container`}>
      <i className={`fa fa-${props.type} icon`} aria-hidden="true" />
    </div>
  )
}
export default Icon
