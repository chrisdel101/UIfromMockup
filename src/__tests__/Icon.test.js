import React from 'react'
import Icon from '../components/utilityComponents/Icon/Icon.jsx'
import renderer from 'react-test-renderer'

test('create and match snapshot', () => {
  const component = renderer.create(
    <Icon src="http://place-puppy.com/200x200" alt="a placeholder" />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
