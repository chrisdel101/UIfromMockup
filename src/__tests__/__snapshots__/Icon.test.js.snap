// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create and match snapshot 1`] = `
<div
  className="icon-container-undefined icon-container"
>
  <i
    aria-hidden="true"
    className="fa fa-undefined icon"
  />
</div>
`;
